packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -3203,20 +3203,34 @@ export class ThinEngine {
         }
     }
 
+    private _args: any[] = [];
+
+    // type def for typescript only
+    private _setUsingContext(functionName: string, arrayLength: number, uniform: Nullable<WebGLUniformLocation>, ...values: any[]): boolean;
+    private _setUsingContext(): boolean {
+        const a = arguments;

```

Comment:
This will be slower right? there is for loop and several tests

We cannot do that so deep in the engine

Reply:
The only thing added was the for, but it can be done using a push instead. the tests all existed before.
This is just a suggestion; I don't mind closing this.

Reply:
It depends. For setInt for instance it is a loss

I would not recommend touching these functions honestly if you don't mind

Reply:
The code for setInt for instance would be:

```
_setUsingContext() {
        const a = arguments;
        for (let i = 2; i < a.length; ++i) {
            this._args[i - 2] = a[i];
        }
        if (!a[2] || (a[1] && (a[1] === 1 ? !a[3].length : a[3] % a[1]))) {
            return false;
        }
        // Matrix functions need false as 2nd variable
        if (a[0].includes("Matrix")) {
            this._args.unshift(this._args.shift(), false);
        }
        this._gl[`uniform${a[0]}`].apply(this._gl, this._args);
        return true;
    }

_setUsingContext();
```

instead of 

```
if (!uniform) {
  return false;
}

this._gl.uniform1i(uniform, value);
```

I am not sure as well the cost is worth the size reduction here as it looks like the function would lose about 10% perf on a quick benchmark

Reply:
It's already closed